File: Python/Creational/Factory-Method/VehicleFactory.py
from Car import Car
from Bike import Bike
from Vehicle import Vehicle


# The only purspose of this class is to get the vehicle class object (Car/Bike) based on the input string.
# So we are keeping the getVehicle method as static method so that we can call it without creating the object of the VehicleFactory class.

class VehicleFactory:
    @staticmethod
    def getVehicle(VehicleType: str) -> Vehicle:
        if VehicleType == "Car":
            vehicle = Car()
        elif VehicleType == "Bike":
            vehicle = Bike()
        else:
            print("Invalid vehicle type")
            exit(1)
        return vehicle

File: CSharp/Creational/Factory-Method/VehicleFactory.cs
using System;

namespace FactoryMethod;

// The only purspose of this class is to get the vehicle class object (Car/Bike) based on the input string.
// So we are keeping the getVehicle method as static method so that we can call it without creating the object of the VehicleFactory class.

public class VehicleFactory
{
    public static Vehicle getVehicle(string VehicleType)
    {
        Vehicle vehicle;
        if (VehicleType == "Car")
        {
            vehicle = new Car();
        }
        else if (VehicleType == "Bike")
        {
            vehicle = new Bike();
        }
        else
        {
            Console.WriteLine("Invalid Vehicle Type");
            return null;
        }
        return vehicle;
    }
}
